services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -174,29 +174,47 @@ public static AuthorizationTokenService instance() {
     }
 
     private static void fireErrorEvent(EventBuilder event, String error, Exception cause) {
-        event.detail(Details.REASON, cause == null || cause.getMessage() == null ? "<unknown>" : cause.getMessage())
-                .error(error);
+        if (cause instanceof CorsErrorResponseException) {
+            // cast the exception to populate the event with a more descriptive reason
+            CorsErrorResponseException originalCause = (CorsErrorResponseException) cause;
+            event.detail(Details.REASON, originalCause.getErrorDescription() == null ? "<unknown>" : originalCause.getErrorDescription())
+                    .error(error);
+        } else {
+            event.detail(Details.REASON, cause == null || cause.getMessage() == null ? "<unknown>" : cause.getMessage())
+                    .error(error);
+        }
+
         logger.debug(event.getEvent().getType(), cause);
     }
-    
-    public Response authorize(KeycloakAuthorizationRequest request) {
+
+    public Response authorize(KeycloakAuthorizationRequest request, EventBuilder event) {
         if (request == null) {
-            throw new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_GRANT, "Invalid authorization request.", Status.BAD_REQUEST);
+            CorsErrorResponseException invalidAuthorizationRequestException = new CorsErrorResponseException(null, OAuthErrorException.INVALID_GRANT, "Invalid authorization request.", Status.BAD_REQUEST);
```

Comment:
Should create the exception using `reqest.getCors()`.

We don't need `Errors.INVALID_AUTHORIZATION_REQUEST`. But just use `Errors.INVALID_REQUEST`.

Reply:
using request.getCors() to create the exception would lead to a NullPointerException because this code section is executed only if request == null (line 191)

Agreed on using Errors.INVALID_REQUEST